### ISSUE TITLE
Incorrect unused-alias warning for namespaced maps

### ISSUE THREAD

**Issue by fancygits** [clj-kondo/clj-kondo#2190]
[ To keep development of this project going, consider sponsoring. If you are
already a sponsor, thank you! ]

**version**
clj-kondo v2023.09.07

**platform**
Native clj-kondo cli on MacOS

**editor**
VS Code with Calva (although the error happens on the CLI)

**problem**
When requiring using `as-alias` and using that alias in a namespace map, `clj-kondo` is incorrectly flagging a `unused-alias` lint error.

**repro**
```
❯ clj-kondo --config '{:linters {:unused-alias {:level :error}}}' --lint - <<< '(ns foo (:require [bar :as-alias b])) #::b{:baz 1}'
::error file=<stdin>,line=1,col=34::Unused alias: b
linting took 38ms, errors: 1, warnings: 0
```

**config**
`{:linters {:unused-alias {:level :error}}}`

**expected behavior**
When using the ns alias in a namespaced map, there are no `unused-alias` warnings/error.

**Comment by borkdude** [comment]
Thanks, will fix in next release
